python/custreamz/custreamz/kafka.py

```diff
@@ -33,6 +33,14 @@ def __init__(self, kafka_configs):
 
         self.kafka_meta_client = KafkaDatasource(self.kafka_confs)
 
+    def list_topics(self):
+
+        """
+        List the topics associated with the underlying Kafka Broker connection.
```

Comment:
The Confluent library's `list_topic` function appears to return all metadata. Can we do the same?

Reply:
I think @kkraus14 makes a very valid point above ^^. When we spoke the other day that idea was to just send the partitions since all you really need is the len/number of partitions. However if you want *all* the metadata the best thing to do is just use python-confluent-kafka. There is no speedup and or need for us to add this to cudf_kafka. I think it best to just use python-confluent-kafka and us close this. Seems the real need is for something that behaves just like python-confluent-kafka so why not just us it?

Reply:
> idea was to just send the partitions since all you really need is the len/number of partitions.

I think this should be OK. 

Using CK's `list_topics` isn't an option here, because the consumers created in streamz are the `custreamz.kafka` consumers. We shouldn't create an additional CK consumer just for this one call.  

If, however, we do end up needing all the metadata in the future, I agree — we should use CK's `list_topics` for sure. But it would definitely be in a different context to solve an entirely different problem, and I think there's no such problem we want to tackle in the short/medium-term roadmap.

Reply:
The above answer is assuming that it is really difficult to have custreamz.kafka's `list_topics` to match CK's API. If it is just a little more work, then I think it is definitely worth doing like Jarod said.